cdap-app-fabric/src/main/java/io/cdap/cdap/metadata/profile/ProfileMetadataMessageProcessor.java

```diff
@@ -252,6 +261,37 @@ private void removeProfileMetadata(MetadataMessage message) throws IOException {
     }
   }
 
+  private void collectPluginMetadata(ApplicationId applicationId, ApplicationSpecification appSpec,
```

Comment:
there's duplicate code here and AppFabricServer, which is an indication that the logic should live somewhere in the metadata services and not in AppFabricServer.